trpcultivate_germcollection/src/Plugin/TripalImporter/GermplasmRelationshipImporter.php

```diff
@@ -0,0 +1,1193 @@
+            }
+          }
+
+          // If any header-row validators failed, skip validation of the data
+          // rows.
+          if ($row_has_failed === TRUE) {
+            $failed_validator = TRUE;
+            break;
+          }
+        }
+
+        if ($line_no > 1) {
+          // Split line into an array using the delimiter supported by this
+          // importer.
+          $data_row = ImportValidationHelper::splitRowIntoColumns($line, $file_mime_type);
+
+          // Call each validator on this row of the file.
+          foreach ($validators['data-row'] as $validator_name => $validator) {
+            // Set failures for this validator name to an empty array to signal
+            // that this validator has been run, but ONLY if it doesn't exist.
+            // (ie. this validator may have already failed on a previous row, so
+            // we don't want to overwrite previous validation failures.)
+            if (!array_key_exists($validator_name, $failures)) {
+              $failures[$validator_name] = [];
+            }
+            if ($validator_name == 'germplasm_name_exists') {
+              // @todo Validate the organism first, once we have a organism
+              // validator plugin, then only set the organism ID if the organism
+              // is valid.
+              // Organism ID:
+              // If Scientific Name is present, lookup the organism ID
+              // and set it in the validator.
+              if ($data_row[2]) {
+                $organism_id = $this->getOrganismIds($data_row[2]);
+              }
+              else {
+                $organism_id = NULL;
+              }
+              if ($organism_id == NULL) {
+                // Manually set failures array.
+                $failed_validator = TRUE;
+                $failures[$validator_name][$line_no] = [
+                  'case' => 'Unable to lookup germplasm with empty values',
+                  'valid' => FALSE,
+                  'failedItems' => ['empty_cells' => $line_no],
+                ];
+                continue;
+              }
+              $validator->setOrganismID($this->organism_ids[$data_row[2]]);
+            }
+            $result = $validator->validateRow($data_row);
+            // Check if validation failed.
+            if (array_key_exists('valid', $result) && $result['valid'] === FALSE) {
+              $failed_validator = TRUE;
+              $failures[$validator_name][$line_no] = $result;
+            }
+          }
+        }
+      }
+
+      fclose($handle);
+    }
+
+    $validation_feedback = $this->processValidationMessages($failures);
+    $storage = $form_state->getStorage();
+    $storage[self::VALIDATION_RESULT] = $validation_feedback;
+    $form_state->setStorage($storage);
+
+    $submit_form = TRUE;
+
+    foreach ($validation_feedback as $feedback_item) {
+      if (($feedback_item['status'] == 'todo' && $form_values['relationship_toggle']) || $feedback_item['status'] == 'fail') {
```

Comment:
Please add a comment here describing in plain language the cases we are checking for here.